bgpd/bgp_vty.c

```diff
@@ -6163,6 +6168,43 @@ ALIAS_HIDDEN(no_neighbor_route_reflector_client,
 	     NO_STR NEIGHBOR_STR NEIGHBOR_ADDR_STR2
 	     "Configure a neighbor as Route Reflector client\n")
 
+/* optimal-route-reflection Root Routers configuration */
+DEFPY (optimal_route_reflection,
+       optimal_route_reflection_cmd,
+       "[no$no] optimal-route-reflection WORD$orr_group [<A.B.C.D|X:X::X:X>$primary [<A.B.C.D|X:X::X:X>$secondary [<A.B.C.D|X:X::X:X>$tertiary]]]",
+       NO_STR
+       "Create ORR group and assign root router(s)\n"
+       "ORR Group name\n"
+       "Primary Root address\n"
+       "Primary Root IPv6 address\n"
+       "Secondary Root address\n"
+       "Secondary Root IPv6 address\n"
+       "Tertiary Root address\n"
+       "Tertiary Root IPv6 address\n")
+{
+	if (!no && !primary) {
+		vty_out(vty, "%% Specify Primary Root address\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+	return bgp_afi_safi_orr_group_set_vty(
+		vty, bgp_node_afi(vty), bgp_node_safi(vty), orr_group,
+		primary_str, secondary_str, tertiary_str, !no);
+}
+
+/* neighbor optimal-route-reflection group*/
+DEFPY (neighbor_optimal_route_reflection,
+       neighbor_optimal_route_reflection_cmd,
+       "[no$no] neighbor <A.B.C.D|X:X::X:X|WORD>$neighbor optimal-route-reflection WORD$orr_group",
+       NO_STR
+       NEIGHBOR_STR
+       NEIGHBOR_ADDR_STR2
+       "Apply ORR group configuration to the neighbor\n"
+       "ORR group name\n")
+{
+	return peer_orr_group_set_vty(vty, neighbor, bgp_node_afi(vty),
+				      bgp_node_safi(vty), orr_group, !no);
```

Comment:
Ditto here about `!no`.

Reply:
!no works and i can use bool set.
anyways will change it to !!no and bool unset in function definition instead of set.